source/source.go

```diff
@@ -0,0 +1,143 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/conduitio-labs/conduit-connector-firebolt/client"
+	"github.com/conduitio-labs/conduit-connector-firebolt/config"
+	"github.com/conduitio-labs/conduit-connector-firebolt/repository"
+	"github.com/conduitio-labs/conduit-connector-firebolt/source/iterator"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+// Iterator defines an Iterator interface needed for the Source.
+type Iterator interface {
```

Comment:
Firebolt client - it is struct what work with Firebolt APi. Firebolt API we use for queries , because firebolt doesn't have db driver for golang  and for starting database engine.
Repository is struct for run queries it is uses  firebolt client. In future if firebolt add golang db driver we can change client to driver.
Iterator use  repository for get rows and iterating.

We can remove repository and use client in iterator. What do you think about it?

Reply:
It sound a like a good idea to me. Feel free to think more about it and/or try it out and let me know what you think about it.